src/main/java/com/epam/processor/AccidentDBServiceImpl.java

```diff
@@ -1,38 +1,92 @@
 package com.epam.processor;
 
+import java.time.LocalTime;
 import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
 
-import com.epam.data.RoadAccident;
+import org.hsqldb.lib.Iterator;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.support.GenericXmlApplicationContext;
+
+import com.epam.dbrepositories.AccidentRepository;
 import com.epam.dbservice.AccidentService;
+import com.epam.entities.Accident;
 
 public class AccidentDBServiceImpl implements AccidentService {
+	
+	@Autowired
+	private AccidentRepository accidentRepository;
 
-	public RoadAccident findOne(String accidentId) {
-		// TODO Auto-generated method stub
+	public Accident findOne(String accidentId) {
+		Iterable<Accident> accidents = accidentRepository.findAll();
+		Iterator iter = (Iterator) accidents.iterator();
+		while(iter.hasNext()) {
+			Accident accident = (Accident) iter.next();
+			if(accidentId.equals(accident.getAccidentIndex())) {
+				System.out.println("Find by accidentId=" + accident);
+				return accident;
+			}
+		}

```

Comment:
try to use java8 code in filtering as we did in 1st homework, It will be more interesting to see. This code is also fine. :+1: